networkx/classes/digraph.py

```diff
@@ -970,7 +970,34 @@ def in_edges(self):
         in_edges : InEdgeView
             A view of edge attributes, usually it iterates over (u, v)
             or (u, v, d) tuples of edges, but can also be used for
-            attribute lookup as `edges[u, v]['foo']`.
+            attribute lookup as `edges[u, v]['foo']`. The *EdgeView family
+            of objects handle requests for all edges without any data. The
+            *EdgeDataView family of objects handle special requests: either
+            restricting to source nodes via nbunch or asking for edge data
+             via data. Also note the difference between G.edges and G.edges()
+             in the examples. G.edges reports edge tuples. If they are multiedges,
+             then the edge tuples are 3-tuples: (n, nbr, edgekey). But if they are
+             not multiedges they are 2-tuples: (n, nbr). But G.edges() returns 2-tuples
+             whether they are multiedges or not.
+
+        Examples
+        -----
+            >>> g = nx.DiGraph()
+            >>> g.add_edge(1,2,color='blue')
+            >>> g.in_edges() # == g.in_edges(nbunch=None)
+            InEdgeView([(1, 2)])
+            >>> g.in_edges(2) # == g.in_edges(nbunch=2)
+            InEdgeDataView([(1, 2)])
+            >>> G = nx.MultiGraph([(1,2,0, {"color": "blue"})])
+            >>> G.edges()
+            MultiEdgeDataView([(1, 2, 0)])
+            >>> list(G.edges())
+            [(1, 2)]
+            >>> G.edges
+            MultiEdgeView([(1, 2, 0)])
+            >>> list(G.edges)
+            [(1, 2, 0)]
```

Comment:
Could you use a capital `G` instead of `g` while creating a graph object?

And also add spaces after a comma (generally follow PEP8).
```
g.add_edge(1,2,color='blue')
```
should be
```
G.add_edge(1, 2, color='blue')
```

Reply:
@MridulS , I have removed the space and renamed g to G

Thanks alot

Reply:
@rossbar can you review this!